source/PlayerInfo.cpp

```diff
@@ -2332,12 +2332,26 @@ void PlayerInfo::UpdateAutoConditions(bool isBoarding)
 		conditions["passenger space"] = flagship->Cargo().BunksFree();
 	}
 	
+	// Clear any existing flagship system: and planet: conditions. (Note: '!' = ' ' + 1.)
+	auto firstS = conditions.lower_bound("flagship system: ");
```

Comment:
We do a similar thing above - write a lambda to do this and pass the condition prefix to it in each case.